.github/workflows/dotnet-build-and-test.yml

```diff
@@ -0,0 +1,43 @@
+name: Run .NET Build and Test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    env:
+      NUGET_PACKAGES: ${{ github.workspace }}/.github/nuget-packages
+    
+    steps:
+      - name: 📝 Fetch Sources 📝
+        uses: actions/checkout@v2
+
+      - name: ⚙ Setup .NET 6.0 SDK ⚙
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
+
+      - name: ⚙ Setup .NET Core 3.1.0 SDK ⚙
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+
+      - name: ♻ NuGet Cache ♻
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.NUGET_PACKAGES }}
+          key: ${{ runner.os }}.nuget.${{ hashFiles('**/*.csproj') }}
+
+      - name: 🔄 Restore Nuget Packages 🔄
```

Comment:
We're using NUKE to build our project (https://www.nuke.build/index.html). Restore, build and test can be replaced with a call to `build.sh` and the `Test` target (https://github.com/JetBrains/space-dotnet-sdk/blob/main/build/Build.cs#L74)

Reply:
Hmm that's quite interesting, thanks for introducing Nuke to me :)

Reply:
Thanks for the update!